arkane/statmech.py

```diff
@@ -347,6 +339,15 @@ def load(self, pdep=False, plot=False):
             statmechLog = determine_qm_software(os.path.join(directory, statmechLog.path))
         else:
             statmechLog.path = os.path.join(directory, statmechLog.path)
+        try:
+            geomLog = local_context['geometry']
+            if isinstance(geomLog, Log) and not isinstance(energy, (GaussianLog, QChemLog, MolproLog)):
```

Comment:
Haha. Not from this PR, but definitely written by me. This checks that the log file is of the `Log` class (which doesn't contain usable methods) but not of any of its descended classes (which has specific methods). If this is the case, then we have to determine which software generated the log file and create that new object ( which is what the following line does).

Do you know if there's a better way to check that something is an instance of a class but not of any of its descendant classes? I didn't find a way when writing this code.

Reply:
Gotcha. Maybe `isinstane(geomLog, Log) and not issubclass(geomLog, Log)` (I haven't tested)? We'll soon incorporate (at least plan to) support for TeraChem and Orca, let's minimize the places we need to modify when adding new software support.

Reply:
Excellent suggestion. I implemented this in all 6 places it occurs in statmech.py, and made a separate commit, because it impacts sections not previously touched with this PR.

Reply:
I changed this from the original implementation because `issubclass` requires both attributes to be classes (not instances) and since a class will return itself as true:

```
import numpy as np

issubclass(np.ndarray, np.ndarray)
Out[2]: True
```

The new implementation uses `isinstance (geomLog, Log) and type(geomLog).__name__ != 'Log'`. This should do the same thing we originally wanted to use `issubclass` for.